docs/config/accessibility.md

```diff
@@ -10,118 +10,169 @@
 
 Now with Mermaid library in much wider use, we have started to work towards more accessible features, based on the feedback from the community.
 
-To begin with, we have added a new feature to Mermaid library, which is to support accessibility options, **Accessibility Title** and **Accessibility Description**.
+Adding accessibility means that the rich information communicated by visual diagrams can be made available to those using assistive technologies (and of course to search engines).
+[Read more about Accessible Rich Internet Applications and the W3 standards.](https://www.w3.org/WAI/standards-guidelines/aria/)
 
-This support for accessibility options is available for all the diagrams/chart types. Also, we have tired to keep the same format for the accessibility options, so that it is easy to understand and maintain.
+Mermaid will automatically insert the [aria-roledescription](#aria-roledescription) and, if provided in the diagram text by the diagram author, the [accessible title and description.](#accessible-title-and-description)
 
-## Defining Accessibility Options
+### aria-roledescription
 
-### Single line accessibility values
+The [aria-roledescription](https://www.w3.org/TR/wai-aria-1.1/#aria-roledescription) for the SVG HTML element is set to the diagram type key. (Note this may be slightly different than the keyword used for the diagram in the diagram text.)
 
-The diagram authors can now add the accessibility options in the diagram definition, using the `accTitle` and `accDescr` keywords, where each keyword is followed by `:` and the string value for title and description like:
+For example: The diagram type key for a state diagram is "stateDiagram". Here (a part of) the HTML of the SVG tag that shows the automatically inserted aria-roledscription set to "stateDiagram". _(Note that some of the SVG attributes and the SVG contents are omitted for clarity.):_
 
-- `accTitle: "Your Accessibility Title"` or
-- `accDescr: "Your Accessibility Description"`
+```html
+<svg
+  aria-roledescription="stateDiagram"
```

Comment:
Is it worth adding a `rendered with Mermaid.js` to make it more clear on what it is?

As a side-note, some of the ARIA specs say:

> User agents MUST NOT expose the `aria-roledescription` property if any of the following conditions exist:
>
> 1. The element to which `aria-roledescription` is applied does not have a valid WAI-ARIA `role` or does not have an implicit WAI-ARIA role semantic.

Do we know if we need to to set a `role="img"` to the `<svg>`, or is this already set?
See https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/img_role#svg_and_roleimg

(if yes, we should add it to the example)

Although note that adding `role="img"` to the `<svg>` means that the screen reader won't show anything within the SVG, I'm unsure if this includes the `<desc>` and the `<title>` elements (although it seems silly if it would ignore them!). I'm also unsure if people would prefer to be able to read through the Mermaid diagrams (garbled as they may be), or if they'd just want to skip over them.

Reply:
Good catch.  <svg> has a default role: "graphics-document"  
[@brennanyoung looked into this in the original issue:](https://github.com/mermaid-js/mermaid/issues/3626#issuecomment-1274324004)

> ...SVG maps to the "graphics-document" role by default, so (in theory at least) we do not need to add role="graphics-document" to the SVG because it's implicit.

So for now we shouldn't set role for svg

Reply:
I don't think "rendered with Mermaid.js" is necessary in the description.  
People viewing a rendered diagram within a webpage won't see and don't care how the diagram was rendered.

Reply:
> Good catch. has a default role: "graphics-document"

The `graphics-document` role looks perfect! Looks like it doesn't appear in https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/document_role because it's an SVG exclusive role, see https://www.w3.org/TR/graphics-aria-1.0/#graphics-document :)

> I don't think "rendered with Mermaid.js" is necessary in the description.

:+1: True! It might help for programmers, since then they know that there would be source code somewhere, but I guess programmers are probably a minority of people visiting each website!